user_group_popover: Fix misaligned icons in user group member list.

sayamsamal opened the pull request:
Previously, due the uneven width between the icons and the user presence indicator, the icons and presence indicator were misaligned with each other.

This PR uses the grid and subgrid layout systems to fix this misalignment issue.

> [!NOTE]  
> Also contains a follow-up commit fixing the extra gap when the group description is absent.

Fixes: [CZO Discussion](https://chat.zulip.org/#narrow/channel/9-issues/topic/bots.20misaligned.20in.20group.20modal/near/1962348)

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**

| Before | After |
|--------|--------|
| ![Screenshot 2024-11-12 at 7 17 31 PM](https://github.com/user-attachments/assets/a88a6e5f-2d2d-41c4-9b2f-c871af588b5a) | ![Screenshot 2024-11-12 at 8 01 41 PM](https://github.com/user-attachments/assets/5ca335b5-6a33-4b04-832c-99fdde91bd4f) | 

**Updated After Screenshot**:
![Screenshot 2024-12-13 at 6 39 36 PM](https://github.com/user-attachments/assets/797f446f-be08-46e4-8d9b-8666e077b40e)


<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

sayamsamal commented:
@karlstolley could you take a look at this?

timabbott commented:
In the "after" screenshot, "Group settings" looks misaligned... but I think the problem may be that we don't want quite that much space between the labels and icons? Not sure, but they look a bit disconnected.

karlstolley commented:
I've basically got this fixed in #32306, including alignment with the Groups settings label (though I should double-check that the bot icon aligns correctly there, too).

timabbott commented:
So is the state that we're waiting for #32306 to setting before looking at this further, as it might be obviated by that?

karlstolley commented on web/styles/popovers.css:
```diff
@@ -337,24 +337,42 @@ ul.popover-group-menu-member-list {
+        display: grid;
+        grid-template-columns:
+            var(--user-group-popover-horizontal-padding)
+            [group-member-icon-start] minmax(5px, auto)
```
It might be better to figure out what the widest icon is, and express that maximum values as ems at 16. `auto` makes me a bit hesitant in this context, but it might be okay.

sayamsamal replied:
The idea of using `grids` with `subgrids` was to make the icons column dynamic, where the largest icon would define the width of that column, and the other smaller icons would center align with respect to that width. So, setting a fixed value defeats that behavior?

sayamsamal replied:
But understanding your concern with using `auto`, which could sometimes behave unexpectedly.  I have instead used `max-content` as the width of the given grid column which should be the ideal property for the given condition?

karlstolley commented on web/styles/popovers.css:
```diff
@@ -337,24 +337,42 @@ ul.popover-group-menu-member-list {
+            var(--user-group-popover-horizontal-padding)
+            [group-member-icon-start] minmax(5px, auto)
+            [group-member-icon-end] var(--user-group-popover-icon-text-gap)
+            [group-member-name-start] 1fr [group-member-name-end] var(
```
`1fr` should be expressed as `minmax(0, 1fr)` to prevent grid blowouts.

While the inline labels are OK here, can you switch this up to use named areas? I generally only use this style of labeling when something needs to occupy a different set of rows/columns from the basic grid.

karlstolley replied:
It might also be worth thinking about whether that padding at the beginning and end belongs as part of the grid, or should sit outside of it. I don't have an opinion, but it seems worth asking (null areas like that can be named with `.`, should you decide they remain)

sayamsamal replied:
Switched to `grid-template-areas` and `grid-template-columns`:
```css
grid-template-areas: "group-member-icon gap group-member-name";
grid-template-columns: max-content var(--user-group-popover-icon-text-gap) minmax(0, 1fr);
```

sayamsamal replied:
I also moved the padding to the `popover-group-menu-member` element itself, as your note about the padding made sense. Defining the padding as null areas in the grid didn't add any advantage to us, so it doesn't make sense to define the same, which we would do via a simple padding to the element itself.

karlstolley commented on web/styles/popovers.css:
```diff
@@ -337,24 +337,42 @@ ul.popover-group-menu-member-list {
+
+    .popover-group-member-icon {
+        justify-self: center;
+        grid-column: group-member-icon-start / group-member-icon-end;
```
You should be able to just reference `group-member-icon` here, and below. That's the magic of the -start and -end on inline references. Though again, I think this would read better with `grid-template-areas:`

sayamsamal replied:
Fixed, as we now just use `grid-area: group-member-icon` and `grid-area: group-member-name`.

sayamsamal commented:
## Update
1. Fixed the space between label and the icons.
![Screenshot 2024-12-13 at 6 39 36 PM](https://github.com/user-attachments/assets/e946a81f-3335-49f1-a7ca-20d07ecfd558)
2. Addressed the feedback from @karlstolley on using grid areas and the need of null areas vs padding.

sayamsamal commented:
@karlstolley, could you please review this PR again?

karlstolley commented:
@sayamsamal can you rebase so I can re-review this?

sayamsamal commented:
@karlstolley rebased to remove the merge conflicts. This is ready for a review.

timabbott commented:
Looks good to me, merged, thanks @sayamsamal! @karlstolley should still feel free to reply or add follow-up tweaks, but I think we're better off merging this rather than risking it sitting over the holidays.
